### PR TITLE
Troubleshooting Windows Builds 2023-11-24

### LINKED ISSUE #781: Compilation with GCC 13
Hi there! 
Arch updated to GCC 13 a couple of weeks ago, and I finally took the plunge.
```
[evert@Evert ~]$ gcc --version
gcc (GCC) 13.1.1 20230429
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Unfortunately, vegastrike-engine-git does not build with this flavor of GCC, and there are a LOT of compilation warnings. 
Sorting those out might get rid of some of the bugs of the game, who knows?

Anyways, it fails to compile in this part:

```
[ 45%] Building CXX object CMakeFiles/vegastrike-engine.dir/src/star_system.cpp.o
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:146,
                 from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Stdafx.h:48,
                 from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/cont_terrain.h:28,
                 from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/star_system.cpp:52:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_Collider.h:198:33: warning: ‘virtual void Opcode::Collider::InitQuery()’ was hidden [-Woverloaded-virtual=]
  198 |     virtual inline_        void InitQuery() {
      |                                 ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:154:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_TreeCollider.h:272:10: note:   by ‘void Opcode::AABBTreeCollider::InitQuery(const Opcode::Matrix4x4*, const Opcode::Matrix4x4*)’
  272 |     void InitQuery(const Matrix4x4 *world0 = nullptr, const Matrix4x4 *world1 = nullptr);
      |          ^~~~~~~~~
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_Collider.h:198:33: warning: ‘virtual void Opcode::Collider::InitQuery()’ was hidden [-Woverloaded-virtual=]
  198 |     virtual inline_        void InitQuery() {
      |                                 ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:158:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_RayCollider.h:257:10: note:   by ‘bool Opcode::RayCollider::InitQuery(const Opcode::Ray&, const Opcode::Matrix4x4*, uint32_t*)’
  257 |     bool InitQuery(const Ray &world_ray, const Matrix4x4 *world = nullptr, uint32_t *face_id = nullptr);
      |          ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:150:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_VolumeCollider.h:128:40: warning: ‘virtual void Opcode::VolumeCollider::InitQuery()’ was hidden [-Woverloaded-virtual=]
  128 |     override(Collider) inline_    void InitQuery() {
      |                                        ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:162:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_SphereCollider.h:93:10: note:   by ‘bool Opcode::SphereCollider::InitQuery(Opcode::SphereCache&, const Opcode::Sphere&, const Opcode::Matrix4x4*, const Opcode::Matrix4x4*)’
   93 |     bool InitQuery(SphereCache &cache,
      |          ^~~~~~~~~
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_VolumeCollider.h:128:40: warning: ‘virtual void Opcode::VolumeCollider::InitQuery()’ was hidden [-Woverloaded-virtual=]
  128 |     override(Collider) inline_    void InitQuery() {
      |                                        ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:166:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_OBBCollider.h:137:10: note:   by ‘bool Opcode::OBBCollider::InitQuery(Opcode::OBBCache&, const Opcode::OBB&, const Opcode::Matrix4x4*, const Opcode::Matrix4x4*)’
  137 |     bool InitQuery(OBBCache &cache,
      |          ^~~~~~~~~
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_VolumeCollider.h:128:40: warning: ‘virtual void Opcode::VolumeCollider::InitQuery()’ was hidden [-Woverloaded-virtual=]
  128 |     override(Collider) inline_    void InitQuery() {
      |                                        ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:170:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_AABBCollider.h:81:10: note:   by ‘bool Opcode::AABBCollider::InitQuery(Opcode::AABBCache&, const Opcode::CollisionAABB&)’
   81 |     bool InitQuery(AABBCache &cache, const CollisionAABB &box);
      |          ^~~~~~~~~
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_VolumeCollider.h:128:40: warning: ‘virtual void Opcode::VolumeCollider::InitQuery()’ was hidden [-Woverloaded-virtual=]
  128 |     override(Collider) inline_    void InitQuery() {
      |                                        ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:174:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_LSSCollider.h:92:10: note:   by ‘bool Opcode::LSSCollider::InitQuery(Opcode::LSSCache&, const Opcode::LSS&, const Opcode::Matrix4x4*, const Opcode::Matrix4x4*)’
   92 |     bool InitQuery(LSSCache &cache,
      |          ^~~~~~~~~
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_VolumeCollider.h:128:40: warning: ‘virtual void Opcode::VolumeCollider::InitQuery()’ was hidden [-Woverloaded-virtual=]
  128 |     override(Collider) inline_    void InitQuery() {
      |                                        ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/Opcode.h:178:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/OPC_PlanesCollider.h:117:10: note:   by ‘bool Opcode::PlanesCollider::InitQuery(Opcode::PlanesCache&, const Opcode::Plane*, uint32_t, const Opcode::Matrix4x4*)’
  117 |     bool InitQuery(PlanesCache &cache, const Plane *planes, uint32_t nb_planes, const Matrix4x4 *worldm = nullptr);
      |          ^~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collide2/CSopcodecollider.h:41,
                 from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/cont_terrain.h:29:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/gfx/mesh.h:362:18: warning: ‘virtual void Mesh::ProcessDrawQueue(size_t, int, bool, const QVector&)’ was hidden [-Woverloaded-virtual=]
  362 |     virtual void ProcessDrawQueue(size_t whichpass, int whichdrawqueue, bool zsort, const QVector &sortctr);
      |                  ^~~~~~~~~~~~~~~~
In file included from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/star_system.cpp:69:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/gfx/sphere.h:115:18: note:   by ‘virtual void SphereMesh::ProcessDrawQueue(int, int, bool, const QVector&)’
  115 |     virtual void ProcessDrawQueue(int whichpass, int which, bool zsort, const QVector &sortctr);
      |                  ^~~~~~~~~~~~~~~~
In file included from /usr/include/c++/13.1.1/ext/alloc_traits.h:34,
                 from /usr/include/c++/13.1.1/bits/stl_list.h:61,
                 from /usr/include/c++/13.1.1/list:65,
                 from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/cmd/collection.h:44,
                 from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/star_system.h:27,
                 from /mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/star_system.cpp:25:
/usr/include/c++/13.1.1/bits/alloc_traits.h: In instantiation of ‘struct std::__allocator_traits_base::__rebind<aligned_allocator<Particle>, Particle, void>’:
/usr/include/c++/13.1.1/bits/alloc_traits.h:94:11:   required by substitution of ‘template<class _Alloc, class _Up> using std::__alloc_rebind = typename std::__allocator_traits_base::__rebind<_Alloc, _Up>::type [with _Alloc = aligned_allocator<Particle>; _Up = Particle]’
/usr/include/c++/13.1.1/bits/alloc_traits.h:228:8:   required by substitution of ‘template<class _Alloc> template<class _Tp> using std::allocator_traits< <template-parameter-1-1> >::rebind_alloc = std::__alloc_rebind<_Alloc, _Tp> [with _Tp = Particle; _Alloc = aligned_allocator<Particle>]’
/usr/include/c++/13.1.1/ext/alloc_traits.h:126:65:   required from ‘struct __gnu_cxx::__alloc_traits<aligned_allocator<Particle>, Particle>::rebind<Particle>’
/usr/include/c++/13.1.1/bits/stl_vector.h:88:21:   required from ‘struct std::_Vector_base<Particle, aligned_allocator<Particle> >’
/usr/include/c++/13.1.1/bits/stl_vector.h:425:11:   required from ‘class std::vector<Particle, aligned_allocator<Particle> >’
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/gfx/particle.h:59:56:   required from here
/usr/include/c++/13.1.1/bits/alloc_traits.h:70:31: error: static assertion failed: allocator_traits<A>::rebind_alloc<A::value_type> must be A
   70 |                         _Tp>::value,
      |                               ^~~~~
/usr/include/c++/13.1.1/bits/alloc_traits.h:70:31: note: ‘std::integral_constant<bool, false>::value’ evaluates to false
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/star_system.cpp: In member function ‘void StarSystem::Update(float)’:
/mnt/int/system/aur/vegastrike-engine-git/src/Vega-Strike-Engine-Source/engine/src/star_system.cpp:1069:11: warning: unused variable ‘normal_simulation_atom’ [-Wunused-variable]
 1069 |     float normal_simulation_atom = SIMULATION_ATOM;
      |           ^~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/vegastrike-engine.dir/build.make:622: CMakeFiles/vegastrike-engine.dir/src/star_system.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:272: CMakeFiles/vegastrike-engine.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: vegastrike-engine-git-exit status 4
 -> Failed to install the following packages. Manual intervention is required:
vegastrike-engine-git - exit status 4

```

### LINKED ISSUE #813: vcpkg error on Windows: `Unrecognized identifer [sic] name xbox`
The full error message is:

```
Fetching registry information from https://github.com/microsoft/vcpkg (HEAD)...
error: Unrecognized identifer name xbox. Add to override list in triplet file.
```

followed by several more repetitions of that last error line.

See, for example, the build output here: https://github.com/vegastrike/Vega-Strike-Engine-Source/actions/runs/6975677063/job/18983122107

Looks like what we are running into is this issue: https://github.com/microsoft/vcpkg/issues/30122

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [In Progress] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues:
Closes #813 
Fixes #781 for 0.9.x and following

Purpose:
- What is this pull request trying to do? Fix the Windows CI builds. Also reduce the number of CI builds overall.
- What release is this for? 0.9.x
- Is there a project or milestone we should apply this to? 0.9.x

### CODE REVIEW

**Comment by stephengtuggy** [comment]
These 7 CI failures were more or less expected. We don't have a Mantic build image yet. And a 25% success rate for macOS is about as good as we've been able to achieve lately, for whatever reason. (Why 25%? Why not 0% or 100%? This needs further investigation.)

File: script/bootstrap [hunk 1]
```diff
@@ -356,20 +348,30 @@ function bootstrapOnUbuntu()
...
                       lsb-release \
                       python3-pip
       python3 -m pip install --upgrade-strategy eager --upgrade pip
       python3 -m pip install --upgrade-strategy eager cmake
       ;;
+    "impish")
+      echo "Sorry, Ubuntu impish is no longer supported"
+      exit 2
+      ;;
+    "hirsute")
```
Comment (BenjamenMeyer) [inline script/bootstrap:371]: FYI the `case` functionality can do multi-match via the pipe operator:

```bash
case $X in

"value1"|"value2"|"value3")
    ...
    ;;
"value4"|"value5")
   ...
   ;;
...
esac
```

**Comment by BenjamenMeyer** [comment]
mantic image should be available; I merged the PR so it's just a matter of that getting through

**Comment by stephengtuggy** [comment]
Okay, so the Rocky Linux and Fedora 38 failures were NOT expected. It looks like I'm running into https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/781 here, as well.

**Comment by stephengtuggy** [comment]
Sweet! Looks like the latest fix for Issue 781 is working! (I mostly borrowed the solution from a couple of other people on the Web who were running into similar issues.)

**Comment by stephengtuggy** [comment]
@royfalk I added you as a reviewer in case you're interested in taking a look. Among other things, I'm cutting way back on the number of CI jobs here. I think it was up to over 60 at one point. Now it's down to 31. A bit more manageable. 😄

**Comment by stephengtuggy** [comment]
Specifically, I derived my fix from https://github.com/nlohmann/json/pull/3895 and https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=64c986b49558a7

Thanks, guys.

**Review by BenjamenMeyer — APPROVED** [review]
I haven't play tested this, but LGTM

File: script/bootstrap (hunk 1 above)
Comment (stephengtuggy) [reply script/bootstrap:371]: So you would like to combine some of these `case` clauses that essentially do the same thing?

**Pull request merged**